modin/config/envvars.py

```diff
@@ -775,16 +791,24 @@ def _sibling(cls) -> type[EnvWithSibilings]:
 )
 
 
-class RangePartitioning(EnvironmentVariable, type=bool):
+def use_range_partitioning_groupby() -> bool:
```

Comment:
the technique of making `RangePartitioning` and `RangePartitioningGroupby` siblings (as we [did it before with `ExperimentalGroupbyImpl`](https://github.com/modin-project/modin/blob/717a32836e6ef1f18f6138f8e6c6939ea301c9cb/modin/config/envvars.py#L749-L752)) won't work here, since we don't want for them to actually share the same value (if they would be siblings, `RangePartitioningGroupby.put(True)` would enable other range-part implementations, not only groupby) that's why introducing this hacky helper function for now